api/tokens/serializers.py

```diff
@@ -1,15 +1,44 @@
 from rest_framework import serializers as ser
 from rest_framework import exceptions
 
-from framework.auth.oauth_scopes import public_scopes
-from osf.models import ApiOAuth2PersonalToken
+from osf.models import ApiOAuth2PersonalToken, ApiOAuth2Scope
 
-from api.base.serializers import JSONAPISerializer, LinksField, IDField, TypeField
+from api.base.serializers import JSONAPISerializer, LinksField, IDField, TypeField, RelationshipField, StrictVersion
+
+
+class TokenScopesRelationshipField(RelationshipField):
+
+    def to_internal_value(self, data):
+        return {'scopes': data}
 
 
 class ApiOAuth2PersonalTokenSerializer(JSONAPISerializer):
     """Serialize data about a registered personal access token"""
 
+    def __init__(self, *args, **kwargs):
+        super(ApiOAuth2PersonalTokenSerializer, self).__init__(*args, **kwargs)
```

Comment:
Some API trickery to add scopes/owner as relationship field for versions 2.11 and greater, and a charfield for 2.10 and lower.  Not only does the field need to be versioned, but the behavior needs to be versioned. Making scopes always embeddable happens here as well, since `get_serializer_context` which makes this possible is called before this serializer is initialized.

This was needed instead of something like ShowIfVersion because in addition to the API changes, the model itself was changed, moving scopes from a charfield to an m2m.  Also, we can't have two fields with the same name, and can't display both fields simultaneously unless we change the name of one of them.

I briefly investigated adding a VersionedField (credit @sloria) or similar, but it seemed like a much longer path to get this uncommon behavior generalized
```
VersionedField({
    '2.11': RelationshipField(....),
    'default': StringField(...),
})
```

Reply:
I agree. To me, this is quite a smart approach.